diff --git a/modules/skparagraph/src/Run.h b/modules/skparagraph/src/Run.h
index f5b3bf57e6..2bc816a1a1 100644
--- a/modules/skparagraph/src/Run.h
+++ b/modules/skparagraph/src/Run.h
@@ -474,7 +474,7 @@ public:
     }
 
     SkScalar height() const {
-        return ::round((double)fDescent - fAscent + fLeading);
+        return fDescent - fAscent + fLeading;
     }
 
     void update(SkScalar a, SkScalar d, SkScalar l) {
