diff --git a/modules/skparagraph/src/Decorations.cpp b/modules/skparagraph/src/Decorations.cpp
index 4239b52316..79d428c921 100644
--- a/modules/skparagraph/src/Decorations.cpp
+++ b/modules/skparagraph/src/Decorations.cpp
@@ -165,7 +165,7 @@ void Decorations::calculatePosition(TextDecoration decoration, SkScalar ascent)
                fFontMetrics.fUnderlinePosition > 0) {
             fPosition  = fFontMetrics.fUnderlinePosition;
           } else {
-            fPosition = fThickness;
+            fPosition = fFontMetrics.fXHeight / 4;
           }
           fPosition -= ascent;
           break;
