diff --git a/src/pdf/SkPDFDocument.cpp b/src/pdf/SkPDFDocument.cpp
index ac82078af7..62d8742c14 100644
--- a/src/pdf/SkPDFDocument.cpp
+++ b/src/pdf/SkPDFDocument.cpp
@@ -334,6 +334,26 @@ static SkPDFIndirectReference append_destinations(
     return doc->emit(destinations);
 }
 
+
+void GetLinkAnnotationConfiguration(sk_sp<SkData> data, SkString* destination, SkString* contents) {
+    if (!data || data->size() == 0) {
+        return;
+    }
+
+    auto annotationData = ToValidUtf8String(*data);
+
+    const auto delimiter = "::questpdf_delimiter::";
+    auto delimiterPosition = annotationData.find(delimiter);
+
+    if (delimiterPosition == -1) {
+        *destination = annotationData;
+        return;
+    }
+
+    *destination = SkString(annotationData.c_str(), delimiterPosition);
+    *contents = SkString(annotationData.c_str() + delimiterPosition + strlen(delimiter));
+}
+
 std::unique_ptr<SkPDFArray> SkPDFDocument::getAnnotations() {
     std::unique_ptr<SkPDFArray> array;
     size_t count = fCurrentPageLinks.size();
@@ -345,14 +365,22 @@ std::unique_ptr<SkPDFArray> SkPDFDocument::getAnnotations() {
     for (const auto& link : fCurrentPageLinks) {
         SkPDFDict annotation("Annot");
         populate_link_annotation(&annotation, link->fRect);
+
+        SkString destination; 
+        SkString contents; 
+        GetLinkAnnotationConfiguration(link->fData, &destination, &contents);
+
+        if (!contents.isEmpty())
+            annotation.insertByteString("Contents", contents);
+
         if (link->fType == SkPDFLink::Type::kUrl) {
             std::unique_ptr<SkPDFDict> action = SkPDFMakeDict("Action");
             action->insertName("S", "URI");
             // This is documented to be a 7 bit ASCII (byte) string.
-            action->insertByteString("URI", ToValidUtf8String(*link->fData));
+            action->insertByteString("URI", destination);
             annotation.insertObject("A", std::move(action));
         } else if (link->fType == SkPDFLink::Type::kNamedDestination) {
-            annotation.insertName("Dest", ToValidUtf8String(*link->fData));
+            annotation.insertName("Dest", destination);
         } else {
             SkDEBUGFAIL("Unknown link type.");
         }
