diff --git a/src/pdf/SkPDFDocument.cpp b/src/pdf/SkPDFDocument.cpp
index 0ecba5f72c..baf8460e83 100644
--- a/src/pdf/SkPDFDocument.cpp
+++ b/src/pdf/SkPDFDocument.cpp
@@ -118,7 +118,7 @@ static_assert((SKPDF_MAGIC[3] & 0x7F) == "Skia"[3], "");
 #endif
 static void serializeHeader(SkPDFOffsetMap* offsetMap, SkWStream* wStream) {
     offsetMap->markStartOfDocument(wStream);
-    wStream->writeText("%PDF-1.4\n%" SKPDF_MAGIC "\n");
+    wStream->writeText("%PDF-1.7\n%" SKPDF_MAGIC "\n");
     // The PDF spec recommends including a comment with four
     // bytes, all with their high bits set.  "\xD3\xEB\xE9\xE1" is
     // "Skia" with the high bits set.
diff --git a/src/pdf/SkPDFMetadata.cpp b/src/pdf/SkPDFMetadata.cpp
index 38438414d8..41a454d922 100644
--- a/src/pdf/SkPDFMetadata.cpp
+++ b/src/pdf/SkPDFMetadata.cpp
@@ -250,7 +250,7 @@ SkPDFIndirectReference SkPDFMetadata::MakeXMPObject(
             " xmlns:xmpMM=\"http://ns.adobe.com/xap/1.0/mm/\"\n"
             " xmlns:pdf=\"http://ns.adobe.com/pdf/1.3/\"\n"
             " xmlns:pdfaid=\"http://www.aiim.org/pdfa/ns/id/\">\n"
-            "<pdfaid:part>2</pdfaid:part>\n"
+            "<pdfaid:part>3</pdfaid:part>\n"
             "<pdfaid:conformance>B</pdfaid:conformance>\n"
             "%s"  // ModifyDate
             "%s"  // CreateDate
